workshops/rust_text_game/README.md

```diff
@@ -0,0 +1,338 @@
+
+    - `use std::io;`
+    
+    Importing libraries gives us the opportunity to expand our programs however you can imagine. By importing `std::io`, we gain access to more functions that allow us to perform "input/output" operations (like `println!()`, but more advanced!).
+
+    - `fn prompt() -> bool {`
+
+    All functions are meant to return some output, which is either visible to the player, to other programs or parts of the same program.
+
+    In this case, we define a new function called `prompt()`, which will return a value known as a `bool`, which stands for "boolean". Boolean data types can only accept two values, which are denoted with the words `true` or `false` in the Rust programming language. They're named after the mathematician George Boole and are possibly the simplest instance of [Boolean algebra](https://en.wikipedia.org/wiki/Boolean_algebra) in the field of computer science.
+
+    - `let mut input = String::new();`
+
+    Here, we create a *mutable* `String`. *Mutability* defines whether you can change the value of a variable. Since we're *initializing* a new string (which means that we declare that the variable `input` exists and is a `String`, but there haven't been any values that have been *declared* to it yet.)
+
+    - `io::stdin().read_line(&mut input).unwrap();`
+
+    Here, using the `io` library again, we capture whatever the player is typing until they hit `Enter` inside the `input` variable.
+
+    - `return input.to_ascii_lowercase().starts_with("y");`
+
+    And finally, we check if an all-lowercase counterpart of the input's value starts with `y`. If it does, it means the input was yes (or similar), and returns `true`. If it doesn't, it's safe to assume that the player replied negatively. Therefore, we return the value`false`.
+
+### Part 4.3: Checking the answer!
+
+In order to write a story, we'll have our program react based on the answers that the player gives. In other words, this is the part where you write your own game's logic! Inside the `intro` function, and under the `println!` macro (= a function whose name ends with `!`), add the following:
+
+```rust
+let answer = prompt();
+```
+
+You now have an (immutable) variable called `answer`, whose type is a `bool`ean (either `true` or `false`). We can now check the player's answer and take a different course of action accordingly:
+
+Let's evaluate the answer that the player gave us!
+
+```rust
+// If the player's answer was yes, print a message
+if answer == true {
+    println!("Yay! Your aspirations will give you some very good karma later in your life.");
+// else, if the answer is no (or anything else!)
+// print a different, less friendly message
+} else {
+    println!("Boo, you're such a buzzkill!");
+}
+```
+
+You can now create a game by similarly adding more prompts, then using them in the `if... else` blocks of the previous prompt to connect them.
+
+### Part 4.4: Writing a full game
+
+Here's a summary of how the code should look like:
```

Comment:
This could introduce it as like handy the project to you... like a DIY thing.. apply your knowledge to make a full thing